### PR TITLE
fix: Wrong status shown in the Comparison View Page of duplicate record

### PR DESCRIPTION
**Pull request by Siyasanga**
https://github.com/opencrvs/opencrvs-core/issues/7439

### CODE REVIEW

File: packages/client/src/views/RegisterForm/duplicate/DuplicateFormTabs.tsx [hunk 1]
```diff
@@ -702,7 +700,11 @@ export const DuplicateFormTabs = (props: IProps) => {
           leftValue: (
             <Text variant="reg16" element="span" color="grey600">
               {actualRegData.status
-                ? intl.formatMessage(regStatusMessages[actualRegData.status])
+                ? intl.formatMessage(
+                    regStatusMessages[
+                      actualRegData.status as unknown as RegStatus
+                    ]
```
Comment (naftis) [inline packages/client/src/views/RegisterForm/duplicate/DuplicateFormTabs.tsx:706]: I prefer changing the status's original type to `RegStatus` [here](https://github.com/opencrvs/opencrvs-core/blob/d46f47e0dbe6cafacd5a03d4d7df6b198849190d/packages/client/src/declarations/index.ts#L228) instead of this type assertion.

You'll need to change https://github.com/opencrvs/opencrvs-core/blob/d46f47e0dbe6cafacd5a03d4d7df6b198849190d/packages/client/src/declarations/index.ts#L1559-L1563 these lines as well to something like 
```
      const downloadedAppStatus: RegStatus | undefined =
        eventData &&
        eventData.registration &&
        eventData.registration.status &&
        eventData.registration.status[0].type
```

, but it would be a nice improvement to the technical debt we have in the front end.

**Review by naftis — CHANGES_REQUESTED** [review]

File: packages/client/src/views/RegisterForm/duplicate/DuplicateFormTabs.tsx (hunk 1 above)
Comment (Siyasanga) [reply packages/client/src/views/RegisterForm/duplicate/DuplicateFormTabs.tsx]: Thanks man, I wasn't sure whether to avoid touching other part of the code as much as possible but I get your point, thinking we could deal with the tech debt as a separate ticket, would you agree?

I'm honestly a bit confused regarding `downloadedAppStatus: RegStatus | undefined` does this mean the type of `downloadedAppStatus` is of `RegStatus`  always? I'm asking because I saw the [DOWNLOAD_STATUS enum](https://github.com/opencrvs/opencrvs-core/blob/1854f16ae65f9308cb4caf5753fcace65a061e13/packages/client/src/declarations/index.ts#L152-L160) with the values below, are they not related?
```
export enum DOWNLOAD_STATUS {
  READY_TO_DOWNLOAD = 'READY_TO_DOWNLOAD',
  DOWNLOADING = 'DOWNLOADING',
  DOWNLOADED = 'DOWNLOADED',
  FAILED = 'FAILED',
  FAILED_NETWORK = 'FAILED_NETWORK',
  READY_TO_UNASSIGN = 'READY_TO_UNASSIGN',
  UNASSIGNING = 'UNASSIGNING'
}
```
I'm avoiding a case of effort spent on refactors bloating up the effort required to do the fix.

Comment (naftis) [reply packages/client/src/views/RegisterForm/duplicate/DuplicateFormTabs.tsx]: >Thanks man, I wasn't sure so I avoided touching other parts of the codebase as much as possible but I get your point, my thinking was, we could deal with the tech debt as a separate ticket. I'm avoiding a case of effort spent on refactors bloating up the effort required to do the fix. would you agree?

Thanks for explaining your thought process, I see you 👍🏻 In our team refactors in pull requests like these are good, unless it's specifically an urgent bug fix. In my opinion, feel free to always apply the 1% improvement to the pull requests :)

File: packages/client/src/views/RegisterForm/review/ReviewSection.tsx [hunk 2]
```diff
@@ -1857,8 +1857,8 @@ class ReviewSectionComp extends React.Component<FullProps, State> {
                     action={
                       viewRecord ||
                       isDuplicate ||
-                      declaration.registrationStatus ===
-                        SUBMISSION_STATUS.CORRECTION_REQUESTED ? null : (
```
Comment (Siyasanga) [inline packages/client/src/views/RegisterForm/review/ReviewSection.tsx:1861]: @naftis This is an example of the type fields being compared

File: CHANGELOG.md [hunk 3]
```diff
@@ -206,6 +205,12 @@ Follow the descriptions in the migration notes to re-provision all servers safel
 - Fix frontend crashing on 'Registration by Status' under performance due to missing translations [#7129](https://github.com/opencrvs/opencrvs-core/pull/7129)
 - Fix email of practitioner to be saved in hearth. A migration is added to correct the email of practitoiner in existing db. [7315](https://github.com/opencrvs/opencrvs-core/pull/7315)
 - Fix inaccessible and only partly visible "Edit" button in "Advanced Search" - feature's date range list [7485](https://github.com/opencrvs/opencrvs-core/pull/7485)
+- Fix wrong status shown in the Comparison View page of the duplicate record [#7439](https://github.com/opencrvs/opencrvs-core/issues/7439)
+
+## Refactor
+
+- Remove dependency on openhim. The openhim db is kept for backwards compatibility reasons and will be removed in v1.6. It has brought some major changes
+  in how the microservices are communicating among them. More on this can be found on the updated [sequence diagrams](https://github.com/opencrvs/opencrvs-core/tree/develop/sequence-diagrams/backend)
```
Comment (Zangetsu101) [inline CHANGELOG.md:213]: This doesn't seem related?

File: packages/client/src/views/RegisterForm/RegisterForm.tsx [hunk 4]
```diff
@@ -314,7 +314,7 @@ function FormAppBar({
     }
     const [exitModalTitle, exitModalDescription] =
       isCorrection(declaration) ||
-      declaration.registrationStatus === SUBMISSION_STATUS.CORRECTION_REQUESTED
+      declaration.submissionStatus === SUBMISSION_STATUS.CORRECTION_REQUESTED
```
Comment (Zangetsu101) [inline packages/client/src/views/RegisterForm/RegisterForm.tsx:317]: This would probably change how the application behaves so it should rather be like this:
```suggestion
      declaration.registrationStatus === RegStatus.CORRECTION_REQUESTED
```

File: packages/client/src/views/RegisterForm/RegisterForm.tsx [hunk 5]
```diff
@@ -451,7 +451,7 @@ function FormAppBar({
               <>
                 {!duplicate &&
                   !isCorrection(declaration) &&
-                  declaration.registrationStatus !==
+                  declaration.submissionStatus !==
                     SUBMISSION_STATUS.CORRECTION_REQUESTED && (
```
Comment (Zangetsu101) [inline packages/client/src/views/RegisterForm/RegisterForm.tsx:455]:
```suggestion
                  declaration.registrationStatus !==
                    RegStatus.CORRECTION_REQUESTED && (
```

File: packages/client/src/views/RegisterForm/duplicate/DuplicateFormTabs.tsx [hunk 6]
```diff
@@ -666,9 +666,7 @@ export const DuplicateFormTabs = (props: IProps) => {
       }
 
       const actualRegData = {
-        status: (props.declaration.data.history as unknown as History[]).find(
-          (data) => data.action === null
-        )?.regStatus,
+        status: props.declaration.registrationStatus,
```
Comment (Zangetsu101) [inline packages/client/src/views/RegisterForm/duplicate/DuplicateFormTabs.tsx:669]: Should we not do the same for `duplicateRegData.status` as well?

File: packages/client/src/views/RegisterForm/review/ReviewSection.tsx [hunk 7]
```diff
@@ -1804,8 +1804,8 @@ class ReviewSectionComp extends React.Component<FullProps, State> {
                             label={sec.title}
                             action={
                               sec.action &&
-                              declaration.registrationStatus !==
-                                SUBMISSION_STATUS.CORRECTION_REQUESTED && (
+                              (declaration.registrationStatus as string) !==
+                                (SUBMISSION_STATUS.CORRECTION_REQUESTED as string) && (
```
Comment (Zangetsu101) [inline packages/client/src/views/RegisterForm/review/ReviewSection.tsx:1808]: I think we can do the same here, basically use `RegStatus`

File: packages/client/src/views/RegisterForm/review/ReviewSection.tsx (hunk 2 above)
Comment (Zangetsu101) [reply packages/client/src/views/RegisterForm/review/ReviewSection.tsx:1861]: I left a couple of comments regarding this, see if they help

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (Siyasanga) [reply CHANGELOG.md:213]: Oh this was me resolving a merge conflict, The refactor changelog entry was from `develop` which was conflicting with my bug changelog entry. I will look into it to see if I can resolve it differently

File: packages/client/src/views/RegisterForm/duplicate/DuplicateFormTabs.tsx [hunk 8]
```diff
@@ -647,8 +647,7 @@ export const DuplicateFormTabs = (props: IProps) => {
       )
 
       const duplicateRegData = {
-        status: eventData.history.find((data: History) => data.action === null)
-          .regStatus as RegStatus,
+        status: props.declaration.registrationStatus,
```
Comment (Zangetsu101) [inline packages/client/src/views/RegisterForm/duplicate/DuplicateFormTabs.tsx:650]: This value needs to be read from `eventData` as this the other(duplicate) declaration. While the `props.declaration` contains the actual declaration we are comparing against

File: packages/client/src/views/RegisterForm/review/ReviewSection.tsx (hunk 7 above)
Comment (Zangetsu101) [inline packages/client/src/views/RegisterForm/review/ReviewSection.tsx:1808]:
```suggestion
                              declaration.registrationStatus !==
                                RegStatus.CorrectionRequested && (
```

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

File: packages/client/src/views/RegisterForm/review/ReviewSection.tsx (hunk 7 above)
Comment (Siyasanga) [reply packages/client/src/views/RegisterForm/review/ReviewSection.tsx:1808]: Thanks @Zangetsu101, I'm not sure how I missed this feedback

File: packages/client/src/views/RegisterForm/duplicate/DuplicateFormTabs.tsx (hunk 8 above)
Comment (Zangetsu101) [reply packages/client/src/views/RegisterForm/duplicate/DuplicateFormTabs.tsx:650]: @Siyasanga How about this one?

**Comment by Siyasanga** [comment]
@Zangetsu101 thanks for catching [this](https://github.com/opencrvs/opencrvs-core/pull/7476#discussion_r1742012962), I've addressed it now

**Review by Zangetsu101 — APPROVED** [review]
